### ISSUE TITLE
Fix `length-zero-no-unit` false positives for custom properties

### ISSUE THREAD

**Issue by mbehzad** [stylelint/stylelint-config-standard#274]
### What steps are needed to reproduce the bug?

Hi everyone,

CSS code:

```scss
.element {
  --offset: 0px;

  left: calc(var(--offset) + 50%);
}
```

stylelintrc config:

```json
{
  "extends": "stylelint-config-standard"
}
```

Error:
```
2:14  ✖  Unexpected unit  length-zero-no-unit
```

Where `left: calc(var(--offset) + 50%);` with `--offset: 0` (without unit) results to invalid `left: calc(0 + 50%);` css.
(mini repo to reproduce: https://github.com/mbehzad/reproduce-length-zero-no-unit-custom-properties-issue)

I think it is more safe to have the rule as 
```
"length-zero-no-unit": [
      true,
      {
        "ignore": ["custom-properties"]
      }
    ],
```

If this makes sense, I can open a PR.
Cheers

### What did you expect to happen?

Not removing the unit for this case: (`--offset: 0px;`)

### What actually happened?

`--offset: 0px;`  needs to be changed to `--offset: 0;`

**Comment by ybiquitous** [comment]
@mbehzad Thanks for opening the issue with the minimal reproduction. I can confirm the behavior you reported.

Considering the background (https://github.com/stylelint/stylelint/issues/2586) where the [`ignore: ["custom-properties"]`](https://stylelint.io/user-guide/rules/length-zero-no-unit#ignore-custom-properties) option was introduced, your proposal sounds to make sense.

**Comment by jeddy3** [comment]
Turning on the [`ignore: ["custom-properties"]`](https://stylelint.io/user-guide/rules/length-zero-no-unit#ignore-custom-properties)` option SGTM. It'll make the config more permissive, so "patch" semver.
